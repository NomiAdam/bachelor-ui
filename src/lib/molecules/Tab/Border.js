import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { findIndex, length, equals, __, divide, multiply, compose } from 'ramda';
import { basicTheme } from '../../constants/theme';
import { minTablet, maxTablet } from '../../constants/device';

const StyledBorder = styled.div`
  position: absolute;
  background-color: ${({ color }) => color};
  transition: 1s all ease;
  @media ${minTablet} {
    height: 2px;
    width: ${({percentage}) => percentage}%;
    bottom: 0;
    left: ${({leftPosition}) => leftPosition}%;
  }
  @media ${maxTablet} {
    width: 2px;
    height: ${({percentage}) => percentage}%;
    left: 0;
    top: ${({leftPosition}) => leftPosition}%;
  }
`;

const multiplyByHundred = multiply(100);
const divideHundred = divide(100);
const findActiveIndex = (active) => findIndex(equals(active));

const Border = ({ items, active, color }) => {
    const lengthOfItems = length(items);
    const percentageOfBorder = compose(
        multiplyByHundred,
        divide(__, lengthOfItems),
        findActiveIndex(active)
    );
    return (
        <StyledBorder
            leftPosition={percentageOfBorder(items)}
            percentage={divideHundred(lengthOfItems)}
            color={color}
        />
    );
};

Border.propTypes = {
    /**
     * Array of strings is expected, ['String', 'Another String'],
     */
    items: PropTypes.array,
    /**
     * String definition of active currently active item
     */
    active: PropTypes.string,
    /**
     * String definition of border color
     */
    color: PropTypes.string,
};

Border.defaultProps = {
    items: [],
    percentage: '0',
    color: basicTheme.purple,
};

export { StyledBorder };
export default Border;
