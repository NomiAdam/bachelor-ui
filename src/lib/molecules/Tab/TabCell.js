import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';

const StyledTabButton = styled.button`
    background-color: inherit;
    position: relative;
    border-color: transparent;
    font-size: .95em;
    text-transform: uppercase;
    color: ${({ active }) => active ? lightTheme.whiteBlue : lightTheme.white};
    outline: none;
    cursor: pointer;
    width: 100%;
    padding: 14px 16px;
    transition: .4s all ease;
    overflow: hidden;
    &:after {
	    content: "";
        background: ${lightTheme.whiteBlue};
        display: block;
        position: absolute;
        padding-top: 350%;
        padding-left: 350%;
        margin-left: -30px !important;
        border-radius: 50%;
        margin-top: -180%;
        opacity: 0;
        transition: all 0.8s
	}
	&:active {
	    &:after {
	        padding: 0;
            margin: 0;
            opacity: 1;
            transition: 0s
	    }
	}
`;

const TabCell = ({handleClick, label, active}) => (
    <StyledTabButton onClick={handleClick(label)} active={active} >{label}</StyledTabButton>
);

TabCell.propTypes = {
    /**
     * String representation of label
     */
    label: PropTypes.string,
    /**
     * onClick action handler
     */
    handleClick: PropTypes.func,
    /**
     * Whether current tab is active or not
     */
    active: PropTypes.bool,
};

TabCell.defaultProps = {
    handleClick: () => () => {},
};

export { StyledTabButton };
export default TabCell
