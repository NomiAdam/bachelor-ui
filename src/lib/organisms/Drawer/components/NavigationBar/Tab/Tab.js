import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import NavigationBar from "../NavigationBar";

export const StyledUl = styled.ul`
	list-style-type: none;
	width: 100%;
    margin: 0;
    padding: 0;
    background-color: #f1f1f1;
`;

const Tab = ({links}) => (
    <StyledUl>
        {links}
    </StyledUl>
);

Tab.propTypes = {
    /**
     * Array of React router Links
     */
    links: PropTypes.any,
};

export default Tab;
