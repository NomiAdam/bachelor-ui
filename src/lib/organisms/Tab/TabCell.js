import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { lightTheme } from '../../constants/theme';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

const StyledTabButton = styled.button`
    background-color: inherit;
    position: relative;
    border-color: ${ resolveTheme(TYPES.COLOR) };
    font-size: .95em;
    text-transform: uppercase;
    color: ${ ({ active, ...props }) => (active ? lightTheme.lightGrey : resolveTheme(TYPES.COLOR)(props)) };
    outline: none;
    cursor: pointer;
    width: 100%;
    padding: 14px 16px;
    transition: .4s all ease;
    overflow: hidden;
    &:after {
	    content: "";
        background: ${ resolveTheme(TYPES.BACKGROUND_COLOR_AFTER) };
        display: block;
        position: absolute;
        padding-top: 350%;
        padding-left: 350%;
        margin-left: -30px !important;
        border-radius: 50%;
        margin-top: -180%;
        opacity: 0;
        transition: all 0.8s
	}
	&:active {
	    &:after {
	        padding: 0;
            margin: 0;
            opacity: 1;
            transition: 0s
	    }
	}
`;

const TabCell = ({
    handleClick, label, active, secondary,
}) => (
    <StyledTabButton secondary={ secondary } onClick={ handleClick(label) } active={ active }>{label}</StyledTabButton>
);

TabCell.propTypes = {
    /**
     * String representation of label
     */
    label: PropTypes.string,
    /**
     * onClick action handler
     */
    handleClick: PropTypes.func,
    /**
     * Whether current tab is active or not
     */
    active: PropTypes.bool,
};

TabCell.defaultProps = {
    handleClick: () => () => {},
};

export { StyledTabButton };
export default TabCell;
