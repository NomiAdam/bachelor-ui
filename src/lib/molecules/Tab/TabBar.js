import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { equals, map } from 'ramda';
import TabCell from './TabCell';
import Border from './Border';
import { lightTheme } from '../../constants/theme';
import { maxTablet } from '../../constants/device';
import DEFAULT_THEME from '../../utils/resolveTheme';

const StyledTabBarWrapper = styled.div`
    ${ DEFAULT_THEME }
    border: none;
    overflow: hidden;
    box-shadow:  0 2px 5px ${ lightTheme.lightGrey };
    display: flex;
    position: relative;
    align-items: center;
    flex-direction: row;
    justify-content: ${ ({ align }) => align };
    @media ${ maxTablet } {
      flex-direction: column;
    }
`;

const TabBar = ({
    handleClick, activeTab, tabOptions, align, border, backgroundColor, borderBorderColor,
}) => {
    const mapTabCell = label => (
        <TabCell
            key={ label }
            label={ label }
            handleClick={ handleClick }
            active={ equals(label, activeTab) }
        />
    );
    const mappedTabCells = map(mapTabCell);
    return (
        <StyledTabBarWrapper align={ align } backgroundColor={ backgroundColor } border={ border }>
            {mappedTabCells(tabOptions)}
            <Border items={ tabOptions } active={ activeTab } color={ borderBorderColor } />
        </StyledTabBarWrapper>
    );
};

TabBar.propTypes = {
    /**
     * onClick action handler, (label) => (e) => {}
     */
    handleClick: PropTypes.func.isRequired,
    /**
     * Array of strings, representing every tab cell, [ 'stringOne', 'stringTwo' ]
     * Note that every string must be unique
     */
    tabOptions: PropTypes.array,
    /**
     * String representation of currently active tab, label of tab is expected
     */
    activeTab: PropTypes.string,
    /**
     * String representation of TabCells align, style according to FlexBox justify-content properties
     */
    align: PropTypes.string,
    /**
     * String definition of border setting
     */
    border: PropTypes.string,
    /**
   * String definition of border Border color
   */
    borderBorderColor: PropTypes.string,
    /**
   * String definition of custom backgroundColor
   */
    backgroundColor: PropTypes.string,
};

TabBar.defaultProps = {
    tabOptions: [],
    align: 'flex-start',
};

export { StyledTabBarWrapper };
export default TabBar;
