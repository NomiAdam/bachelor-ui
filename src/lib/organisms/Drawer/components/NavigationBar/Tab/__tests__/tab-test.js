/* eslint-disable no-undef */
import React from 'react';
import Tab from '../Tab';
import { shallow } from 'enzyme';

describe('Tab', () => {
    it('should be defined', () => {
        expect(Tab).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Tab />)).toMatchSnapshot();
    });
});
