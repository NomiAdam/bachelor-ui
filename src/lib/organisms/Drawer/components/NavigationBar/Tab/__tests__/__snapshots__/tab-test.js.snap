// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should match the snapshot 1`] = `<styled.ul />`;
